src/main/features/response/draft/responseDraft.ts

```diff
@@ -105,6 +106,13 @@ export class ResponseDraft extends DraftDocument {
     return this.isResponsePopulated() && this.response.type === ResponseType.FULL_ADMISSION
   }
 
+  // TODO: Because of an overlap between two stories (ROC-3657, ROC-3658), the logic of this function
+  // is incomplete. ROC-3658 should revisit once 'statement of means' flow is complete.
+  public isResponseFullyAdmittedWithInstalments (): boolean {
+    return this.isResponseFullyAdmitted()
+      && (_.get(this, 'fullAdmission.paymentOption.option') === DefendantPaymentType.INSTALMENTS)
```

Comment:
I am not big fan of this as this text based solution will make refactoring more difficult. As long as I would be keen to use it in JS would I think it works against the benefits of TS.

It also makes impossible jumping to the right place in IDE.

I would strongly suggest changing it to regular TS property access.

Reply:
You made two good point here. Agreed it sort of defeats the purpose of using TS in the first place.

Reply:
Code updated.